arch/xtensa/src/esp32/esp32_wifi_adapter.c

```diff
@@ -1841,7 +1841,7 @@ static int32_t esp_task_get_max_priority(void)
 
 static void *esp_malloc(uint32_t size)
 {
-  return malloc(size);
+  return kmm_malloc(size);
```

Comment:
Could please also change the `_free` member of the `g_wifi_osi_funcs` structure from `free` to `kmm_free`.

There are actually a few other violation of the sort in this file, especially around files management.  We had an internal PR that addressed all of those but it has stalled for other reasons.
Will fix more in the upcoming days.

Reply:
@Ouss4 Done, I left the pthread mutex since it need more work to replace.

Reply:
Thanks

Reply:
@xiaoxiang781216 did you forget to push?

Reply:
Ohh, I forgot push my change, it should OK now.